lib/puppet/provider/package/pip.rb

```diff
@@ -60,7 +61,16 @@ def query
   # cache of PyPI's package list so this operation will always have to
   # ask the web service.
   def latest
-    client = XMLRPC::Client.new2("http://pypi.python.org/pypi")
+    http_proxy_host = Puppet::Util::HttpProxy.http_proxy_host
+    http_proxy_port = Puppet::Util::HttpProxy.http_proxy_port
+    if http_proxy_host and http_proxy_port
```

Comment:
We require the use of logical operators (&&) instead of the and/or operators as because in some instances (not this case, though) the precedence can cause unintended consequences.